### ISSUE TITLE
Experiment: Compare ingestion times with canonical chain discovery vs directly ingestion blocks.

### ISSUE THREAD

**Issue by trevorbernard** [Granola-Team/mina-indexer#1268]
Our canonical chain discovery algorithm takes a non-trivial amount of time to calculate and verify. 25mins with 811K blocks. The hypothesis is the second deserialization costs more then the work savings by ingesting the canonical chain first then the orphaned and recent paths.

Modify the indexer to bypass the canonical chain discovery and start adding the blocks directly. Compare with the current indexer ingestion times on the same blocks.

**Comment by Isaac-DeFrain** [comment]
Any progress on this?

**Comment by trevorbernard** [comment]
No, I made a small attempt but my build had issues so I parked it. It's currently in the `wishful` category.
